mips/gt64120.c

```diff
@@ -0,0 +1,77 @@
+#include <mips/gt64120.h>
+#include <pci.h>
+
+#define PCI0_CFG_ADDR_R GT_R(GT_PCI0_CFG_ADDR)
+#define PCI0_CFG_DATA_R GT_R(GT_PCI0_CFG_DATA)
+
+#define PCI0_CFG_REG_SHIFT 2
+#define PCI0_CFG_FUNCT_SHIFT 8
+#define PCI0_CFG_DEV_SHIFT 11
+#define PCI0_CFG_BUS_SHIFT 16
+#define PCI0_CFG_ENABLE 0x80000000
+
+static unsigned gt_pci_make_addr(pci_device_t *dev, unsigned reg) {
+  return (((dev->addr.device) << PCI0_CFG_DEV_SHIFT) |
+          ((dev->addr.function) << PCI0_CFG_FUNCT_SHIFT) |
+          ((reg) << PCI0_CFG_REG_SHIFT));
+}
+
+typedef union {
+  uint8_t byte[4];
+  uint16_t word[2];
+  uint32_t dword;
+} pci_reg_t;
+
+static uint32_t gt_pci_read_config(pci_device_t *dev, unsigned reg,
```

Comment:
Would it be possible to make this function universal for all `pci_bus`es, not just the particular one called `gt`? I suppose `pci_bus` might contain information about `CFG_ADDR` and `CFG_DATA` registers, and using them would make these two functions bus-independent. Are some other PCI bus implementations configured using different mechanisms (special processor instructions come to mind)?